File: dsa master/linked list/traversal.java
class Node {

    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

public class LinkedListTraversal {

    Node head;

    // Traverse and print the linked list
    void printList() {

        Node current = head;

        while (current != null) {
            System.out.print(current.data + " ");
            current = current.next;
        }

        System.out.println();
    }

    public static void main(String[] args) {

        LinkedListTraversal list = new LinkedListTraversal();

        // Create nodes
        list.head = new Node(10);

        Node second = new Node(20);
        Node third = new Node(30);
        Node fourth = new Node(40);

        // Connect nodes
        list.head.next = second;
        second.next = third;
        third.next = fourth;

        // Traverse the linked list
        System.out.println("Linked List:");

        list.printList();
    }
}

File: dsa master/linked list/traversal.py
class Node:

    def __init__(self, data):
        self.data = data
        self.next = None


def travesal(head):
    current = head

    while current is not None:
        print(current.data, end=" ")
        current = current.next

def main():
    first = Node(10)
    second = Node(20)
    third = Node(30)

    first.next = second
    second.next = third

    head = first

    travesal(head)
    
main()
